### ISSUE TITLE
thnets on Windows

### ISSUE THREAD

**Issue by AlexandrSmirnov** [mvitez/thnets#15]
Hi Marko,
Is it possible to build this to run on Windows 10? I have some troubles with OpenBlas, when I am trying to make it.

**Comment by mvitez** [comment]
As it's now, no. With some modifications, probably yes, I've never tried.

**Comment by AlexandrSmirnov** [comment]
Can you advise, how I need to change makefile or something else, to make it on Windows?

**Comment by mvitez** [comment]
You should use MSYS/MinGW and then start to see the problems and solve them one by one.

**Comment by AlexandrSmirnov** [comment]
Ok, I got this error. Does it means, something wrong with OpenBlas?
make output:
$ mingw32-make
gcc -Wall -c -fPIC -fopenmp -O3 thload.c
gcc -Wall -c -fPIC -fopenmp -O3 thbasic.c
gcc -Wall -c -fPIC -fopenmp -O3 thapi.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/SpatialConvolutionMM.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/SpatialMaxPooling.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/Threshold.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/View.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/SoftMax.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/Linear.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/Dropout.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/SpatialZeroPadding.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/Reshape.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/SpatialConvolution.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/Normalize.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/SpatialFullConvolution.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/SpatialMaxUnpooling.c
modules/SpatialMaxUnpooling.c: In function 'nn_SpatialMaxUnpooling_updateOutput':
modules/SpatialMaxUnpooling.c:110:11: warning: 'owidth' may be used uninitialized in this function [-Wmaybe-uninitialized]
   #pragma omp parallel for private(p)
           ^~~
modules/SpatialMaxUnpooling.c:110:11: warning: 'oheight' may be used uninitialized in this function [-Wmaybe-uninitialized]
gcc -Wall -c -fPIC -fopenmp -O3 modules/SpatialBatchNormalization.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/SpatialAveragePooling.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/Sequential.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/Concat.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/ConcatTable.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/JoinTable.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/CAddTable.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/PReLU.c
gcc -Wall -c -fPIC -fopenmp -O3 pytorch.c
gcc -Wall -c -fPIC -fopenmp -O3 modules/LogSoftMax.c
gcc -Wall -c -fPIC -fopenmp -O3 OpenBLAS-stripped/sgemm.c
OpenBLAS-stripped/sgemm.c: In function 'gemm_thread':
OpenBLAS-stripped/sgemm.c:712:59: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
     job[mypos].working[i][CACHE_LINE_SIZE * bufferside] = (long)buffer[bufferside];
                                                           ^
OpenBLAS-stripped/sgemm.c:732:7: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
       (float *)job[current].working[mypos][CACHE_LINE_SIZE * bufferside], c, ldc, m_from, xxx);
       ^
OpenBLAS-stripped/sgemm.c:96:97: note: in definition of macro 'KERNEL_OPERATION'
 #define KERNEL_OPERATION(M, N, K, ALPHA, SA, SB, C, LDC, X, Y) sgemm_kernel(M, N, K, ALPHA, SA, SB, (float *)(C) + ((X) + (Y) * LDC), LDC)
                                                                                                 ^~
OpenBLAS-stripped/sgemm.c:764:7: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
       (float *)job[current].working[mypos][CACHE_LINE_SIZE * bufferside], c, ldc, is, xxx);
       ^
OpenBLAS-stripped/sgemm.c:96:97: note: in definition of macro 'KERNEL_OPERATION'
 #define KERNEL_OPERATION(M, N, K, ALPHA, SA, SB, C, LDC, X, Y) sgemm_kernel(M, N, K, ALPHA, SA, SB, (float *)(C) + ((X) + (Y) * LDC), LDC)
                                                                                                 ^~
gcc -Wall -c -fPIC -fopenmp -O3 OpenBLAS-stripped/sger.c
gcc -Wall -c -fPIC -fopenmp -O3 OpenBLAS-stripped/sgemv.c
gcc -Wall -c -fPIC -fopenmp -O3 OpenBLAS-stripped/gemm_beta.c
gcc -Wall -c -fPIC -fopenmp -O3 OpenBLAS-stripped/gemv_t.c
gcc -Wall -c -fPIC -fopenmp -O3 OpenBLAS-stripped/copy.c
g++ -o libthnets.so thload.o thbasic.o thapi.o SpatialConvolutionMM.o SpatialMaxPooling.o Threshold.o View.o SoftMax.o Linear.o Dropout.o SpatialZeroPadding.o Reshape.o SpatialConvolution.o Normalize.o SpatialFullConvolution.o SpatialMaxUnpooling.o SpatialBatchNormalization.o SpatialAveragePooling.o Sequential.o Concat.o ConcatTable.o JoinTable.o CAddTable.o PReLU.o pytorch.o LogSoftMax.o sgemm.o sger.o sgemv.o gemm_beta.o gemv_t.o copy.o -shared -fopenmp -lm
sgemm.o:sgemm.c:(.text+0x1794): undefined reference to `sgemm_oncopy'
sgemm.o:sgemm.c:(.text+0x17d2): undefined reference to `sgemm_kernel'
sgemm.o:sgemm.c:(.text+0x18b5): undefined reference to `sgemm_kernel'
sgemm.o:sgemm.c:(.text+0x190b): undefined reference to `sgemm_incopy'
sgemm.o:sgemm.c:(.text+0x1937): undefined reference to `sgemm_itcopy'
sgemm.o:sgemm.c:(.text+0x19fb): undefined reference to `sgemm_incopy'
sgemm.o:sgemm.c:(.text+0x1a7b): undefined reference to `sgemm_otcopy'
sgemm.o:sgemm.c:(.text+0x1ab1): undefined reference to `sgemm_kernel'
sgemm.o:sgemm.c:(.text+0x1af5): undefined reference to `sgemm_itcopy'
sgemm.o:sgemm.c:(.text+0x2022): undefined reference to `sgemm_otcopy'
sgemm.o:sgemm.c:(.text+0x2070): undefined reference to `sgemm_kernel'
sgemm.o:sgemm.c:(.text+0x20f1): undefined reference to `sgemm_oncopy'
sgemm.o:sgemm.c:(.text+0x22c9): undefined reference to `sgemm_kernel'
sgemm.o:sgemm.c:(.text+0x24a5): undefined reference to `sgemm_kernel'
sgemm.o:sgemm.c:(.text+0x263a): undefined reference to `sgemm_incopy'
sgemm.o:sgemm.c:(.text+0x2675): undefined reference to `sgemm_itcopy'
sgemm.o:sgemm.c:(.text+0x2743): undefined reference to `sgemm_incopy'
sgemm.o:sgemm.c:(.text+0x2787): undefined reference to `sgemm_itcopy'
sger.o:sger.c:(.text+0xf2): undefined reference to `saxpy_k'
collect2.exe: error: ld returned 1 exit status
Makefile:188: recipe for target 'libthnets.so' failed
mingw32-make: *** [libthnets.so] Error 1

**Comment by mvitez** [comment]
All the undefined reference errors come from missing assembly files in the Makefile.

**Comment by AlexandrSmirnov** [comment]
Can you contact me on email alessander.smirnov@gmail.com ?, I have commercial proposal for you
